File: app/src/main/java/com/example/osamaomar/shopgate/presentationn/screens/feature/splash/SplashActivity.java
package com.example.osamaomar.shopgate.presentationn.screens.feature.splash;

import android.content.Intent;
import android.os.Handler;
import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;

import com.example.osamaomar.shopgate.R;
import com.example.osamaomar.shopgate.helper.ResourceUtil;
import com.example.osamaomar.shopgate.presentationn.screens.feature.home.mainactivity.MainActivity;

import java.util.ArrayList;

public class SplashActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_splash);

        new Handler().postDelayed(() -> {

            if (ResourceUtil.getCurrentLanguage(SplashActivity.this).matches("ar"))
                ResourceUtil.changeLang("ar",SplashActivity.this);
            else
                ResourceUtil.changeLang("en",SplashActivity.this);

            Intent mainIntent = new Intent(SplashActivity.this, MainActivity.class);

            startActivity(mainIntent);
            finish();
        }, 5000);
    }
}

File: app/src/main/java/com/example/osamaomar/shopgate/presentationn/screens/feature/splash/SplashActivity.kt
package com.example.osamaomar.shopgate.presentationn.screens.feature.splash

import android.content.Intent
import android.support.v7.app.AppCompatActivity
import android.os.Bundle
import android.os.Handler
import com.example.osamaomar.shopgate.R
import com.example.osamaomar.shopgate.presentationn.screens.feature.home.mainactivity.MainActivity

class SplashActivity : AppCompatActivity() {

    private var mDelayHandler: Handler? = null
    private val SPLASH_DELAY: Long = 5000 //5 seconds

    internal val mRunnable: Runnable = Runnable {
        if (!isFinishing) {
            val intent = Intent(applicationContext, MainActivity::class.java)
            startActivity(intent)
            finish()
        }
    }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_splash)

        //Initialize the Handler
        mDelayHandler = Handler()

        //Navigate with delay
        mDelayHandler!!.postDelayed(mRunnable, SPLASH_DELAY)

    }

    public override fun onDestroy() {

        if (mDelayHandler != null) {
            mDelayHandler!!.removeCallbacks(mRunnable)
        }

        super.onDestroy()
    }
}
